frontend/src/scene/main-view/ClassCard.js

```diff
@@ -303,14 +221,20 @@ const Card = class extends React.Component {
                                 >
                                     {buttonLabel}
```

Comment:
Can you change the text here based on the error status.
`errorDetail !== null  ? {seeCourseDetails} : {buttonLabel}`

You can add translation to `content.json
courseCard.seeCourseDetails = "View Details" (Actual translation can be got from Vantaa or Juho later)

Reply:
errorDetail is never null due to `const errorDetail = getErrorDetail(course, errorMessages) || {}`. Removing `|| {}` will make `<CardWrapper {...rest} errorColorCode={errorDetail.colorCode || ''}>` fail to compile. A workaround is using `errorDetail.shortMessage ?`